x/sigs/handler_test.go

```diff
@@ -0,0 +1,131 @@
+package sigs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestBumpSequence(t *testing.T) {
+	var (
+		key1 = weavetest.NewKey().PublicKey()
+		key2 = weavetest.NewKey().PublicKey()
+	)
+
+	cases := map[string]struct {
+		// Before performing the test, initialize the database with given user data.
+		InitData       []*UserData
+		Msg            BumpSequenceMsg
+		WantCheckErr   *errors.Error
+		WantDeliverErr *errors.Error
+		WantSequences  []*UserData
+	}{
+		"great success": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 1},
+				{Pubkey: key2, Sequence: 9},
+			},
+			Msg: BumpSequenceMsg{
+				Pubkey:    key1,
+				Increment: 1,
+			},
+			WantSequences: []*UserData{
+				{Pubkey: key1, Sequence: 2},
+				{Pubkey: key2, Sequence: 9},
+			},
+		},
+		"message with a missing public key is invalid": {
```

Comment:
req: if you come up with a use case then this can still default to the main signer (which is the use case I had in mind)